common/src/main/java/org/opensearch/ml/common/parameter/MLModel.java

```diff
@@ -0,0 +1,141 @@
+        this.name = name;
+        this.algorithm = algorithm;
+        this.version = version;
+        this.content = content;
+        this.user = user;
+    }
+
+    public MLModel(FunctionName algorithm, Model model) {
+        this(model.getName(), algorithm, model.getVersion(), Base64.getEncoder().encodeToString(model.getContent()), null);
+    }
+
+    public MLModel(StreamInput input) throws IOException{
+        name = input.readOptionalString();
+        algorithm = input.readEnum(FunctionName.class);
+        version = input.readInt();
+        content = input.readOptionalString();
+        if (input.readBoolean()) {
+            this.user = new User(input);
+        } else {
+            user = null;
+        }
+    }
+
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(name);
+        out.writeEnum(algorithm);
+        out.writeInt(version);
+        out.writeOptionalString(content);
+        if (user != null) {
+            out.writeBoolean(true); // user exists
+            user.writeTo(out);
+        } else {
+            out.writeBoolean(false); // user does not exist
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (name != null) {
+            builder.field(MODEL_NAME, name);
+        }
+        if (algorithm != null) {
+            builder.field(ALGORITHM, algorithm);
+        }
+        if (version != null) {
+            builder.field(MODEL_VERSION, version);
+        }
+        if (content != null) {
+            builder.field(MODEL_CONTENT, content);
+        }
+        if (user != null) {
+            builder.field(USER, user);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static MLModel parse(XContentParser parser, String taskId) throws IOException {
+        String name = null;
+        FunctionName algorithm = null;
+        Integer version = null;
+        String content = null;
+        User user = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case MODEL_NAME:
+                    name = parser.text();
+                    break;
+                case MODEL_CONTENT:
+                    content = parser.text();
+                    break;
+                case MODEL_VERSION:
+                    version = parser.intValue();
+                    break;
+                case USER:
+                    user = User.parse(parser);
+                    break;
+                case ALGORITHM:
+                    algorithm = FunctionName.valueOf(parser.text());
+                    break;
+                default:
+                    break;
```

Comment:
Add `parser.skipChildren();` before line125 to skip any wrong field name. Refer to AD code https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/model/AnomalyDetector.java#L552